nixos/modules/services/networking/searx.nix

```diff
@@ -3,32 +3,110 @@
 with lib;
 
 let
-
+  dataDir = "/var/lib/searx";
   cfg = config.services.searx;
 
-  configFile = cfg.configFile;
+  # Script to merge NixOS settings with
+  # the default settings.yml bundled in searx.
+  mergeConfig = ''
+    cd ${dataDir}
+    # find the default settings.yml
+    default=$(find '${cfg.package}/' -name settings.yml)
+
+    # write NixOS settings
+    cat <<'EOF' > settings.yml
+      ${generators.toYAML { } cfg.settings}
+    EOF
+
+    # merge the two files
+    ${pkgs.yq-go}/bin/yq m -P -i settings.yml "$default"
```

Comment:
I don't remember the search config fully but is it completely mergeable? Does that do deep merge? What happens if a list has 5 entries and I want t overwrite them?

Reply:
I asked that myself, but surprisingly it's working as expected. In particular, it's merging the `engine` list like a set.
For example, I'm redefining the API keys for a bunch of engines and the result is applying the changes but keeping every other property.

Reply:
I still don't have a good feeling about this but the current way is probably more terrible.

Reply:
Yeah, a bit more testing wouldn't hurt.

Reply:
On a more close inspection I found the bitter truth: it did merge the list of engines but overriding unrelated elements. It looks like it's sorting in alphabetical order by `name` and then merging the respective elements.